### ISSUE TITLE
Potential Error in Test Set: Calculator ID 15 (Child-Pugh Score, Ascites Not Considered)

### ISSUE THREAD

**Issue by FutureForMe** [ncbi-nlp/MedCalc-Bench#8]
Hi~
Thank you for your great work on this dataset.

I noticed that there might be an error in the Test set of the MedCalc dataset. Based on the Ground Truth Explanation field, it seems that for Calculator ID 15 (Child-Pugh Score calculation), Ascites was not considered. For example, in the Test set, the sample with question_id 208 includes Ascites in the Relevant Entities field, but it appears that Ascites was not utilized in the reasoning process.

**Is this behavior reasonable?**

```
"question_id": 208,  
...
"Relevant Entities": "{'Bilirubin': [1.6, 'mg/dL'], 'Ascites': 'slight', 'international normalized ratio': 2.0, 'Albumin': [3.0, 'g/dL']}",
...
"Ground Truth Explanation": "The current child pugh score is 0.\nThe patient's INR is 2.0. Because the INR is between 1.7 and 2.3, we add two to the score, making the current total 0 + 2 = 2.\nThe concentration of bilirubin is 1.6 mg/dL. Because the Bilirubin concentration is less than 2 mg/dL, we add 1 to the score, making the current total 2 + 1 = 3.\nThe concentration of albumin is 3.0 g/dL. Because the Albumin concentration is between 2.8 g/dL and 3.5 g/dL, we add 2 to the score, making the current total 3 + 2 = 5.\nEncephalopathy state is not specified, and so we assume it's value to be 'no encephalopathy.' We add one point to the score, making the current total 5 + 1 = 6.\nThe patient's child pugh score is 6."
```

**Comment by nikhilk7153** [comment]
Hi Haotian, 

You are correct that this is a mistake. What happened is that I did not check the whether the the annotators labels matched in string values that I was using in our implementation. As mentioned in #7, I will resolve this issue along with a few other calculators where there might be a some small issues and release it as a version 1.1., by next Tuesday the latest. Is that ok with you?

**Comment by FutureForMe** [comment]
Looking forward to the latest version!

**Comment by FutureForMe** [comment]
```
"question_id": 6821
...
"Calculator Name": "SIRS Criteria",
...
"Ground Truth Explanation": "The current count of SIRS criteria met is 0.\nThe patient's temperature is 37.4 degrees celsius. Because the temperature is between 36 and 38 degrees celsius, this does not meet SIRS criteria for temperature, and so the current total remains at 0.\nThe patient's heart rate is 77.0 beats per minute. Because the heart rate is less than 90 beats per minute, this does not meet SIRS criteria for heart rate, and so the current total remains at 0.\nThe patient's concentration of white blood cell is 27580.0 count/µL. To convert 27580.0 count/µL of white blood cell to m^3, multiply by the conversion factor 1000000000.0 µL/m^3 which will give 27580.0 white blood cell count/µL * 1000000000.0 µL/m^3 = 27580000000000.0 white blood cell count/m^3. Because the white blood cell count is greater than 12000 count per m^3, we increment the criteria count by 1 making the current total 1 + 1 = 2.\nThe final SIRS criteria is whether the patient has a respiratory rate of more than 20 breaths per minute or if the patient's PaCO₂ partial pressure is less than 32 mm Hg. The patient's respiratory rate is 18.0 breaths per minute, which is less or equal to than 20 breaths per min. The patient's PaCO₂ partial pressure is not provided and so we assume that the patient's partial pressure is greater than or equal to 32 mm Hg.Neither criteria met and so keep the current total at 2.\nHence, the the number of SIRS criteria met by the patient is 2."
```
_The patient's concentration of white blood cell is 27580.0 count/µL. To convert 27580.0 count/µL of white blood cell to m^3, multiply by the conversion factor 1000000000.0 µL/m^3 which will give 27580.0 white blood cell count/µL * 1000000000.0 µL/m^3 = 27580000000000.0 white blood cell count/m^3. Because the white blood cell count is greater than 12000 count per m^3, we increment the criteria count by 1 making the current total 1 + 1 = 2._ 

Could you please check whether the reasoning process for Question ID 6821 in the training set also contains an error?

**Comment by nikhilk7153** [comment]
Yes, this is an error which I am aware about. I have corrected this in an updated version where I manually checked all of the answers with MDCalc.com, but am just going through all of the instances another time for final verification with my annotators. Note that we have made other changes in this version as well such as replacing some notes which we felt might have been ambiguous in their ground truth extractions when having multiple people examine the note as opposed to just one annotator in the initial version. I have also added the full formulas in all of the explanations for all of the calculators in this new version. 

For the time being, if you would like to use this for experimentation, feel free to use this test dataset at: https://docs.google.com/spreadsheets/d/1qUyF6LhJt1fYij3d3KpmRAmB_7maIoBfLKd_cY7CiZs/edit?usp=sharing. I will make this one version 1.1 within the next week and update the repository. 

For the training set, we do not perform such verifications due to cost of annotation, but you can find an updated version here which reflects the changes in the step-by-step explanations: https://docs.google.com/spreadsheets/d/1y36URmNF3t5xlXujjFuGdMbjyk9H9xohiSIB1Jec0oQ/edit?gid=390922739#gid=390922739

Unlike the first instance which involved only one person checking, we are employing four rounds of verification (we are currently on the fourth round, but once this is resolved, I will update the Github/HuggingFace accordingly).

**Comment by FutureForMe** [comment]
Thank you for your response. We will use the latest version of the Test dataset you provided.
